db/migrate/20250214153835_move_blacklisted_events_to_settings_changes.rb

```diff
@@ -0,0 +1,119 @@
+    "ManageIQ::Providers::EmbeddedTerraform::AutomationManager"  => "/ems/ems_embedded_terraform/blacklisted_event_names",
+    "ManageIQ::Providers::Foreman::ConfigurationManager"         => "/ems/ems_foreman/blacklisted_event_names",
+    "ManageIQ::Providers::Google::CloudManager"                  => "/ems/ems_google/blacklisted_event_names",
+    "ManageIQ::Providers::IbmCic::CloudManager"                  => "/ems/ems_ibm_cic/blacklisted_event_names",
+    "ManageIQ::Providers::IbmPowerHmc::InfraManager"             => "/ems/ems_ibm_power_hmc/blacklisted_event_names",
+    "ManageIQ::Providers::IbmPowerVc::CloudManager"              => "/ems/ems_ibm_power_vc/blacklisted_event_names",
+    "ManageIQ::Providers::IbmTerraform::ConfigurationManager"    => "/ems/ems_ibm_terraform/blacklisted_event_names",
+    "ManageIQ::Providers::Kubernetes::ContainerManager"          => "/ems/ems_kubernetes/blacklisted_event_names",
+    "ManageIQ::Providers::Kubevirt::InfraManager"                => "/ems/ems_kubevirt/blacklisted_event_names",
+    "ManageIQ::Providers::Lenovo::PhysicalInfraManager"          => "/ems/ems_lenovo/blacklisted_event_names",
+    "ManageIQ::Providers::Nsxt::NetworkManager"                  => "/ems/ems_nsxt/blacklisted_event_names",
+    "ManageIQ::Providers::Nuage::NetworkManager"                 => "/ems/ems_nuage/blacklisted_event_names",
+    "ManageIQ::Providers::Openshift::ContainerManager"           => "/ems/ems_openshift/blacklisted_event_names",
+    "ManageIQ::Providers::Openstack::CloudManager"               => "/ems/ems_openstack/blacklisted_event_names",
+    "ManageIQ::Providers::OracleCloud::CloudManager"             => "/ems/ems_oracle_cloud/blacklisted_event_names",
+    "ManageIQ::Providers::Ovirt::InfraManager"                   => "/ems/ems_ovirt/blacklisted_event_names",
+    "ManageIQ::Providers::Ovirt::NetworkManager"                 => "/ems/ems_ovirt_network/blacklisted_event_names",
+    "ManageIQ::Providers::Redfish::PhysicalInfraManager"         => "/ems/ems_redfish/blacklisted_event_names",
+    "ManageIQ::Providers::Redhat::InfraManager"                  => "/ems/ems_redhat/blacklisted_event_names",
+    "ManageIQ::Providers::Redhat::NetworkManager"                => "/ems/ems_redhat_network/blacklisted_event_names",
+    "ManageIQ::Providers::Vmware::InfraManager"                  => "/ems/ems_vmware/blacklisted_event_names",
+    "ManageIQ::Providers::Vmware::CloudManager"                  => "/ems/ems_vmware_cloud/blacklisted_event_names",
+    "ManageIQ::Providers::Vmware::NetworkManager"                => "/ems/ems_vmware_cloud_network/blacklisted_event_names",
+    "ManageIQ::Providers::Vmware::ContainerManager"              => "/ems/ems_vmware_tanzu/blacklisted_event_names",
+    "ManageIQ::Providers::Workflows::AutomationManager"          => "/ems/ems_workflows/blacklisted_event_names"
+  }.freeze
```

Comment:
This was a bit interesting.  Because this was using the vendor instead of the ems_type there were some classes (e.g. Amazon::CloudManager, Amazon::NetworkManager, Amazon::ContainerManager) which had duplicate blacklisted events because they were all using the `Settings.ems.ems_amazon.blacklisted_event_names` setting.

I need to have this full list above so I can skip all of these as default, but I don't want or need to replicate that here and this can be a straight map of 1 EMS class to 1 settings key.

I got the list of settings keys by iterating through `Settings.ems.keys` where there was a `Settings.ems.send(key).key?(:blacklisted_event_names)` key.  I then manually assigned what I knew to be the "main" EMS class.

Reply:
Also interesting, there seems to have been some copy&paste going on here as there are a number of providers with a blacklisted_event_names setting but no event catcher.

```
>> PROVIDER_MODEL_TO_SETTINGS_KEY.keys.select { |klass| "#{klass}::EventCatcher".safe_constantize.to_s != "#{klass}::EventCatcher" }
=> 
["ManageIQ::Providers::EmbeddedTerraform::AutomationManager",
 "ManageIQ::Providers::Foreman::ConfigurationManager",
 "ManageIQ::Providers::IbmTerraform::ConfigurationManager",
 "ManageIQ::Providers::Nsxt::NetworkManager",
 "ManageIQ::Providers::Ovirt::NetworkManager",
 "ManageIQ::Providers::Redhat::NetworkManager",
 "ManageIQ::Providers::Vmware::NetworkManager",
 "ManageIQ::Providers::Workflows::AutomationManager"]
```